dashboard/pages/Weekly_Newsletter.py

```diff
@@ -39,13 +39,17 @@ def verify_email(email_address: str, config):
 
 def add_to_database(first_name: str, last_name: str, email: str) -> None:
     '''Adds a row to the subscriber table in our database.'''
+    if email == '':
+        raise ValueError('No email given!')
+
     conn = get_db_connection(ENV)
     with conn.cursor() as cur:
 
         cur.execute("""INSERT INTO subscriber (first_name,last_name,email)
                     VALUES (%s,%s,%s)""", (first_name, last_name, email))
 
     conn.commit()
+    conn.close()
```

Comment:
It may be better to open one connection at the beginning of your main and just use the same open connection, then close it once in main, but its deffo optional and wont affect how your code works. Looks solid otherwise 👍

Reply:
Good point, I think I will leave it how it is for now, but will bear this in mind for the future!